api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts

```diff
@@ -22,7 +23,9 @@ export class SharedStorage implements TemporaryDirectory {
   }
 
   async cleanup(directory: string): Promise<void> {
-    // TODO check if starts with tempDirectory
+    if (this.#hasPoisonNullByte(directory)) {
+      throw new Error(`Hacking is not allowed.`);
+    }
```

Comment:
shouldn't we validate it on input, before cleaning up?

Reply:
@Dyostiq would you mind to elaborate? It is stateless, so you can pass any directory and it validates it on every use 🤔